[Feat] 숙제 CRUD 구현

beomzun opened the pull request:
## 📄 Description

- close : #2

> 숙제 CRUD 구현 

## 📌 구현 내용

- [x] 숙제 생성
- [x] 숙제 조회
- [x] 숙제 수정
- [x] 숙제 삭제
- [x] (추가) BaseEntity status 필드 초기값 설정
- [x] (추가) PR 템플릿 위치 이동 -> 두 개의 .github dir에 들어가있어 반영되지 않았음

## ✅ PR 포인트
- 빌더 패턴 처음 적용해봤습니다
- 영속성 컨텍스트 내부에서 dirty check를 통해 수정로직 구현하였습니다
- 생성 시 응답 헤더에 location 필드로 URI(생성한 id)를 전달합니다

## 📚 TO-DO List
- Test 학습 후 적용

beomzun committed:
feat: BaseEntity status 초기값 설정

beomzun committed:
fix: 컨트롤러 응답 방법 수정

beomzun committed:
fix: 컨트롤러 응답 방법 수정

beomzun committed:
Feat: 숙제 CRUD 구현

숙제 도메인의 CRUD를 빌더 패턴을 적용하여 구현하였음

beomzun committed:
chore: PR 템플릿 위치 이동

2hy2on reviewed (approved):
코드 잘봤습니다. toDetailedDto와 toEntity 메소드는 entity, dto 클래스가 아닌 converter 클래스에 만들어 따로 빼서 구현하는 편이 entity, dto 클래스의 목적을 흐리지 않아 더욱 가독성 높은 클래스 구조라 생각합니다. 이 부분 한번 고려해 보시면 좋을 것 같습니다:)

beomzun committed:
Refactor: Converter 메서드 분리

- 기존의 Dto와 Entity 클래스 내부에 있던 메서드를 Converter 클래스로 분리하여 가독성 향상
- 기존의 toDetailedDto() 에서 불필요하게 인자로 넘기던 homeworkId 삭제

beomzun commented:
converter 메서드 분리하여 적용해봤습니다! 좋은 방법 알려주셔서 감사합니다ㅎㅎ
